src/analysis/normed_space/extend.lean

```diff
@@ -0,0 +1,161 @@
+/-
+Copyright (c) 2020 Ruben Van de Velde All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ruben Van de Velde
+-/
+
+import analysis.complex.basic
+import analysis.normed_space.operator_norm
+
+/-!
+# Extending a continuous ℝ-linear map to a continuous ℂ-linear map
+
+In this file we provide a way to extend a continuous ℝ-linear map to a continuous ℂ-linear map in
+a way that bounds the norm by the norm of the original map.
+
+We motivate the form of the extension as follows. Note that `fc : F →ₗ[ℂ] ℂ` is defined fully by
+`Re fc`: for all `x : F`, `fc (I • x) = I * fc x`, so `Im (fc x) = -Re (fc (I • x))`. Therefore,
+given an `fr : F →ₗ[ℝ] ℝ`, we define `fc x = fr x - fr (I • x) * I`.
+-/
+
+open complex
+
+variables {F : Type*} [normed_group F] [normed_space ℂ F]
+
+/-- Extend `fr : F →ₗ[ℝ] ℝ` to `F →ₗ[ℂ] ℂ` in a way that will also be continuous and have its norm
+bounded by `∥fr∥` if `fr` is continuous. -/
+noncomputable def linear_map.extend_to_ℂ (fr : F →ₗ[ℝ] ℝ) : F →ₗ[ℂ] ℂ :=
+begin
+  let fc : F → ℂ := λ x, ⟨fr x, -fr (I • x)⟩,
+  have add : ∀ x y : F, fc (x + y) = fc x + fc y,
+  {
```

Comment:
I'll fix the brace positions before merging.

Reply:
No need to wait for reviewers to fix formatting issues -- it makes it easier for us if these are out of the way first.